cmd/rpc/sock.go

```diff
@@ -0,0 +1,455 @@
+package rpc
+
+import (
+	"fmt"
+	"github.com/canopy-network/canopy/controller"
+	"github.com/canopy-network/canopy/lib"
+	"github.com/gorilla/websocket"
+	"github.com/julienschmidt/httprouter"
+	"net/http"
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+)
+
+/* This file implements the client & server logic for the 'root-chain info' and corresponding 'on-demand' calls to the rpc */
+
+var _ lib.RCManagerI = new(RCManager)
+
+// RCManager handles a group of root-chain sock clients
+type RCManager struct {
+	c             lib.Config                   // the global node config
+	subscriptions map[uint64]*RCSubscription   // chainId -> subscription
+	subscribers   map[uint64]*RCSubscriber     // chainId -> subscriber
+	l             sync.Mutex                   // thread safety
+	afterRCUpdate func(info lib.RootChainInfo) // callback after the root chain info update
+	upgrader      websocket.Upgrader           // upgrade http connection to ws
+	log           lib.LoggerI                  // stout log
+}
+
+// NewRCManager() constructs a new instance of a RCManager
+func NewRCManager(controller *controller.Controller, c lib.Config, l lib.LoggerI) (man *RCManager) {
+	// create the manager
+	man = &RCManager{
```

Comment:
as we're aiming for understanding above all things I think that some of the naming, although it may seem obvious to most readers, is not clear at a glance ofwhat they do. (for example I was like "what is 'man'?" for a few seconds until I realized is short of manager). Expanding their name a bit could help with that 
```suggestion
func NewRCManager(controller *controller.Controller, config lib.Config, logger lib.LoggerI) (man *RCManager) {
	// create the manager
	manager = &RCManager{
```